src/conn.c

```diff
@@ -3422,6 +3427,26 @@ natsConnection_ConnectTo(natsConnection **newConn, const char *url)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+natsConnection_Reconnect(natsConnection *nc)
+{
+    if (nc == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    natsConn_Lock(nc);
+    if (natsConn_isClosed(nc))
+    {
+        natsConn_Unlock(nc);
+        return nats_setDefaultError(NATS_CONNECTION_CLOSED);
+    }
+
+    nc->forceReconnect = true;
```

Comment:
See my earlier comment. I understand that this function use-case is to refresh auth, but it seems to me a bit weird that since we have an option that explicitly configures the connection to reject reconnect, we would allow a call `natsConnection_Reconnect()` to work and override this option. Meaning that if users wanted to use `natsConnection_Reconnect()` they would probably not use the option to explicitly disallow reconnection.